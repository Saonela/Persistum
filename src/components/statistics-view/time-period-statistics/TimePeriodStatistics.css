
.time-period-statistics__item {
    margin-bottom: var(--margin-l)
}

.time-period-statistics__period {
    font-size: var(--font-size-l);
    text-align: left;
    margin-bottom: var(--margin-m)
}

.time-period-statistics__period-secondary {
    font-size: var(--font-size);
    margin-left: var(--margin-xs);
}

.time-period-statistics__list-header {
    display: flex;
    justify-content: space-between;
    max-width: 740px;
    margin-bottom: var(--margin-m);
    font-size: var(--font-size-m);
}

.time-period-statistics__header-item {
    max-width: 246px;
    width: 100%;
}

.time-period-statistics__list-header *:first-child {
    text-align: left;
}

.time-period-statistics-entry {
    display: flex;
    align-items: center;
    justify-content: space-between;
    max-width: 740px;
    margin-bottom: var(--margin-s);
}

.time-period-statistics-entry:hover {
    color: var(--color-theme);
}

.time-period-statistics-entry__count {
    max-width: 246px;
    width: 100%;
}

@media only screen and (max-width: 540px) {
    .time-period-statistics__list-header {
        display: none;
    }

    .time-period-statistics {
        padding-top: var(--padding-l);
        padding-right: var(--padding-m);
        padding-left: var(--padding-m);
        padding-bottom: var(--margin);
        text-align: left;
    }

    .time-period-statistics-entry {
        flex-direction: column;
        align-items: end;
    }

    .time-period-statistics-entry > *:not(:first-child)::before {
        content: attr(data-label);
        display: inline-block;
        margin-right: var(--margin-xs);
    }

    .time-period-statistics-entry > *:first-child {
        margin-bottom: var(--margin-s);
    }

    .time-period-statistics-entry:not(:last-child) {
        padding-bottom: var(--margin);
        border-bottom: 1px solid var(--color-border);
    }
}
