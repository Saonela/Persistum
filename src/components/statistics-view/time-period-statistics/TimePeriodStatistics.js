import React from "react";
import "./TimePeriodStatistics.css"
import moment from "moment";
import ActivityEntry from "../../shared/activity-entry/ActivityEntry";

function TimePeriodStatisticsPeriod({item}) {
    if (item.type === 'year') {
        return (
            <span>{moment(item.period, 'YYYY-MM').format("YYYY")}</span>
        )
    } else {
        return (
            <>
                <span>{moment(item.period, 'YYYY-MM').format("MMMM")}</span>
                <span className="time-period-statistics__period-secondary" style={{fontSize: '13px'}}>
                    {moment(item.period, 'YYYY-MM').format("YYYY")}
                </span>
            </>
        )
    }
}

function TimePeriodStatistics({activities, statistics}) {
    return (
        <div className="app-panel app-border time-period-statistics statistics-panel">
            <div className="statistics-panel__header">Time Period</div>
            {statistics.map(item =>
                <div key={item.period} className="time-period-statistics__item">
                    <div className="time-period-statistics__period">
                        <TimePeriodStatisticsPeriod item={item}/>
                    </div>
                    <div className="time-period-statistics__list-header">
                        <p className="time-period-statistics__header-item">Activity</p>
                        <p className="time-period-statistics__header-item">Count</p>
                    </div>
                    {activities.map(activity =>
                        <div className="time-period-statistics-entry" key={activity.id}>
                            <ActivityEntry activity={activity}/>
                            <p data-label="Count:" className="time-period-statistics-entry__count">{item.activities[activity.id].count}</p>
                        </div>
                    )}
                </div>
            )}
        </div>
    )
}

export default TimePeriodStatistics;
