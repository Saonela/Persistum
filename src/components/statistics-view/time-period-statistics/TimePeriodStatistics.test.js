import {render, screen} from "@testing-library/react";
import React from "react";
import TimePeriodStatistics from "./TimePeriodStatistics";

const activities = [
    {
        id: 123456,
        name: 'Read 5 pages',
        completed: false,
        style: ''
    },
    {
        id: 999,
        name: 'Workout 30 minutes',
        completed: false,
        style: ''
    }
];

const statistics = [
    {
        period: '2020',
        type: 'year',
        activities: {
            999: {
                count: 3
            },
            123456: {
                count: 1
            }
        }
    },
    {
        period: '2020-04',
        type: 'month',
        activities: {
            999: {
                count: 1
            },
            123456: {
                count: 0
            }
        }
    },
    {
        period: '2020-11',
        type: 'month',
        activities: {
            999: {
                count: 2
            },
            123456: {
                count: 1
            }
        }
    }
];

describe('TimePeriodStatistics', () => {

    it('should display statistics', () => {
        render(<TimePeriodStatistics activities={activities} statistics={statistics}/>);
        expect(screen.getAllByText('Read 5 pages')).toHaveLength(3);
        expect(screen.getAllByText('Workout 30 minutes')).toHaveLength(3);
    });

    it('should match snapshot', () => {
        const {asFragment} = render(<TimePeriodStatistics activities={activities} statistics={statistics}/>);
        expect(asFragment()).toMatchSnapshot();
    });
});
