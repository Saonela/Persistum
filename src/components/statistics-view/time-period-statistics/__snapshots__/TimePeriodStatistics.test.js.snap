// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimePeriodStatistics should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="app-panel app-border time-period-statistics statistics-panel"
  >
    <div
      class="statistics-panel__header"
    >
      Time Period
    </div>
    <div
      class="time-period-statistics__item"
    >
      <div
        class="time-period-statistics__period"
      >
        <span>
          2020
        </span>
      </div>
      <div
        class="time-period-statistics__list-header"
      >
        <p
          class="time-period-statistics__header-item"
        >
          Activity
        </p>
        <p
          class="time-period-statistics__header-item"
        >
          Count
        </p>
      </div>
      <div
        class="time-period-statistics-entry"
      >
        <div
          class="activity-entry"
          data-label="Activity"
        >
          <div
            aria-label="Activity color"
            class="activity-color"
            role="img"
          />
          Read 5 pages
        </div>
        <p
          class="time-period-statistics-entry__count"
          data-label="Count:"
        >
          1
        </p>
      </div>
      <div
        class="time-period-statistics-entry"
      >
        <div
          class="activity-entry"
          data-label="Activity"
        >
          <div
            aria-label="Activity color"
            class="activity-color"
            role="img"
          />
          Workout 30 minutes
        </div>
        <p
          class="time-period-statistics-entry__count"
          data-label="Count:"
        >
          3
        </p>
      </div>
    </div>
    <div
      class="time-period-statistics__item"
    >
      <div
        class="time-period-statistics__period"
      >
        <span>
          April
        </span>
        <span
          class="time-period-statistics__period-secondary"
          style="font-size: 13px;"
        >
          2020
        </span>
      </div>
      <div
        class="time-period-statistics__list-header"
      >
        <p
          class="time-period-statistics__header-item"
        >
          Activity
        </p>
        <p
          class="time-period-statistics__header-item"
        >
          Count
        </p>
      </div>
      <div
        class="time-period-statistics-entry"
      >
        <div
          class="activity-entry"
          data-label="Activity"
        >
          <div
            aria-label="Activity color"
            class="activity-color"
            role="img"
          />
          Read 5 pages
        </div>
        <p
          class="time-period-statistics-entry__count"
          data-label="Count:"
        >
          0
        </p>
      </div>
      <div
        class="time-period-statistics-entry"
      >
        <div
          class="activity-entry"
          data-label="Activity"
        >
          <div
            aria-label="Activity color"
            class="activity-color"
            role="img"
          />
          Workout 30 minutes
        </div>
        <p
          class="time-period-statistics-entry__count"
          data-label="Count:"
        >
          1
        </p>
      </div>
    </div>
    <div
      class="time-period-statistics__item"
    >
      <div
        class="time-period-statistics__period"
      >
        <span>
          November
        </span>
        <span
          class="time-period-statistics__period-secondary"
          style="font-size: 13px;"
        >
          2020
        </span>
      </div>
      <div
        class="time-period-statistics__list-header"
      >
        <p
          class="time-period-statistics__header-item"
        >
          Activity
        </p>
        <p
          class="time-period-statistics__header-item"
        >
          Count
        </p>
      </div>
      <div
        class="time-period-statistics-entry"
      >
        <div
          class="activity-entry"
          data-label="Activity"
        >
          <div
            aria-label="Activity color"
            class="activity-color"
            role="img"
          />
          Read 5 pages
        </div>
        <p
          class="time-period-statistics-entry__count"
          data-label="Count:"
        >
          1
        </p>
      </div>
      <div
        class="time-period-statistics-entry"
      >
        <div
          class="activity-entry"
          data-label="Activity"
        >
          <div
            aria-label="Activity color"
            class="activity-color"
            role="img"
          />
          Workout 30 minutes
        </div>
        <p
          class="time-period-statistics-entry__count"
          data-label="Count:"
        >
          2
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
